app/src/lib/stats/stats-store.ts

```diff
@@ -225,14 +231,46 @@ export class StatsStore {
   }
 
   /** Set whether the user has opted out of stats reporting. */
-  public setOptOut(optOut: boolean) {
+  public setOptOut(optOut: boolean): Promise<void> {
     this.optOut = optOut
 
-    localStorage.setItem('stats-opt-out', optOut ? '1' : '0')
+    localStorage.setItem(StatsOptOutKey, optOut ? '1' : '0')
+
+    return this.sendOptOutStatusPing(optOut)
   }
 
   /** Has the user opted out of stats reporting? */
   public getOptOut(): boolean {
     return this.optOut
   }
+
+  /** Post some data to our stats endpoint. */
+  private post(body: object): Promise<Response> {
+    const options = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(body),
+    }
+
+    return fetch(StatsEndpoint, options)
+  }
+
+  private async sendOptOutStatusPing(optOut: boolean): Promise<void> {
+    const eventType = optOut ? 'opt_out' : 'opt_in'
+    try {
+      const response = await this.post({ eventType })
+      if (!response.ok) {
+        throw new Error(`Unexpected status: ${response.statusText} (${response.status})`)
+      }
+
+      localStorage.setItem(HasSentOptOutStatusKey, '1')
+
+      console.log('Opt out reported.')
```

Comment:
Can we import `logInfo` from `lib/logging/renderer` and use that here instead? It'll automatically log to the console